src/Scorecard.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+
+class Scorecard {
+
+  constructor() {
+    this.STARTING_SCORE = 0;
+    this.score = this.STARTING_SCORE;
+    this.bonusScore = this.STARTING_SCORE
+    this.previousFrame = 0;
+    this.strikesCount = [];
+    this.strikeSum = 0;
+  }
+
+  isPreviousFrameSpare() {
+    return this.previousFrame > 9;
+  }
+
+  calculateStrikeTally(){
+    if (this.strikesCount.length == 1){
+      this.strikeSum += (this.strikesCount[1] + this.strikesCount[2] + 
+      this.strikesCount[3]);
+    }
+    else {
+      for (let i = 0; (i < this.strikesCount.length - 2); i++) {
+        this.strikeSum += (this.strikesCount[i] + this.strikesCount[i + 1] + 
+        this.strikesCount[i + 2]);
+      }
+      }
+  }
+
+  roll(value){
```

Comment:
roll(value) and rollTenthFrame(value) are both pretty long functions, can these be refactored to reduce the length?